llama_index/llms/litellm.py

```diff
@@ -0,0 +1,502 @@
+from typing import Any, Awaitable, Callable, Dict, Optional, Sequence
+
+from llama_index.bridge.pydantic import Field
+
+from llama_index.callbacks import CallbackManager
+from llama_index.llms.base import (
+    LLM,
+    ChatMessage,
+    ChatResponse,
+    ChatResponseAsyncGen,
+    ChatResponseGen,
+    CompletionResponse,
+    CompletionResponseAsyncGen,
+    CompletionResponseGen,
+    LLMMetadata,
+    llm_chat_callback,
+    llm_completion_callback,
+)
+from llama_index.llms.generic_utils import (
+    achat_to_completion_decorator,
+    acompletion_to_chat_decorator,
+    astream_chat_to_completion_decorator,
+    astream_completion_to_chat_decorator,
+    chat_to_completion_decorator,
+    completion_to_chat_decorator,
+    stream_chat_to_completion_decorator,
+    stream_completion_to_chat_decorator,
+)
+from llama_index.llms.openai_utils import (
+    acompletion_with_retry,
+    completion_with_retry,
+    from_openai_message_dict,
+    is_chat_model,
+    is_function_calling_model,
+    openai_modelname_to_contextsize,
+    to_openai_message_dicts,
+    validate_openai_api_key,
+)
+
+
+class LiteLLM(LLM):
+    class_type = "litellm"
+
+    model: str = Field(description="The LiteLLM model to use.") # For complete list of providers https://docs.litellm.ai/docs/providers
+    temperature: float = Field(description="The tempature to use during generation.")
+    max_tokens: Optional[int] = Field(
+        description="The maximum number of tokens to generate."
+    )
+    additional_kwargs: Dict[str, Any] = Field(
+        default_factory=dict, description="Additonal kwargs for the OpenAI API."
+    )
+    max_retries: int = Field(description="The maximum number of API retries.")
+
+    def __init__(
+        self,
+        model: str = "gpt-3.5-turbo",
+        temperature: float = 0.1,
+        max_tokens: Optional[int] = None,
+        additional_kwargs: Optional[Dict[str, Any]] = None,
+        max_retries: int = 10,
+        api_key: Optional[str] = None,
+        api_type: Optional[str] = None,
+        callback_manager: Optional[CallbackManager] = None,
+        **kwargs: Any,
+    ) -> None:
+        validate_openai_api_key(api_key, api_type)
```

Comment:
Doesn't lite-llm support more than openai? Or would this initial version only support openai models?

Reply:
it supports more than openai - thanks for the catch.